src/script/lua_api/l_internal.h

```diff
@@ -43,3 +43,22 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define GET_ENV_PTR         \
 	MAP_LOCK_REQUIRED;      \
 	GET_ENV_PTR_NO_MAP_LOCK
+
+
+/**
+ * This macro permits to bootstrap easily a Lua Referenced object
+ */
+#define LUAREF_OBJECT(name)                                                                                        \
+private:                                                                                                           \
+    name *m_object = nullptr;                                                                                                \
```

Comment:
I'm not quite happy about this name. It makes the readability worse because the variable does no longer describe what kind of object it actually is.